pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -299,24 +302,31 @@ func (c *EtcdProxyController) syncHandler(key string) error {
 	// attempt processing again later. This could have been caused by a
 	// temporary network failure, or any other transient reason.
 	if err != nil {
-		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
-		return err
+		etcdstorageCondition.Status = etcdstoragev1alpha1.ConditionFalse
+		etcdstorageCondition.Reason = "ReplicaSetFailure"
+		etcdstorageCondition.Message = err.Error()
+
+		msg := fmt.Sprintf("Unable to create or get ReplicaSet %s: %v", replicaSetName(etcdstorage), err.Error())
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ReplicaSetCreationFailure, msg)
+
+		errs = append(errs, err)
 	}
 
-	// If the ReplicaSet is not controlled by this EtcdStorage resource, we should log
-	// a warning to the event recorder and ret
+	// If the ReplicaSet is not controlled by this EtcdStorage resource, we should try to update Owner reference.
 	if !metav1.IsControlledBy(replicaset, etcdstorage) {
 		replicaset.SetOwnerReferences([]metav1.OwnerReference{
 			*metav1.NewControllerRef(etcdstorage, etcdstoragev1alpha1.SchemeGroupVersion.WithKind("EtcdStorage")),
 		})
+
 		replicaset, err = c.kubeclientset.AppsV1().ReplicaSets(c.config.ControllerNamespace).Update(replicaset)
 		if err != nil {
-			msg := fmt.Sprintf(MessageErrResourceReclaimed, replicaset.Name)
-			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceReclaimed, msg)
-			return err
```

Comment:
OwnerRef errors are not critical anymore. If something happens, I'm just firing event, but I'm not failing or appending it to errors. Not sure is this a good or bad decision, but my reasoning is that this is not critical, even if it fails, it's somewhat okay, EtcdProxy is deployed.

If this is a bad decision, just appending it to errors is going to fix it.

Reply:
Decided to add it to errors.